### PR TITLE
th/cli-connection-handling

### PR DESCRIPTION
**Pull request by thom311**
The assumption that one connection is only active once in each moment is going to change.
I was looking a bit at nmcli how it handles this, and it was mostly OK already.

Still, this is some rework of `connection.c` that came out of this.

Please review.

### CODE REVIEW

File: clients/cli/connections.c [hunk 1]
```diff
@@ -4738,27 +4724,24 @@ do_connection_add (NmCli *nmc, int argc, char **argv)
...
 
-	/* Create a new connection object */
 	connection = nm_simple_connection_new ();
 
-	/* Build up the 'connection' setting */
 	s_con = (NMSettingConnection *) nm_setting_connection_new ();
 	nm_connection_add_setting (connection, NM_SETTING (s_con));
 
 read_properties:
+	g_clear_error (&error);
```
Comment (fgiudici) [inline clients/cli/connections.c]: g_clear_error()  moved here does not make sense: error will be already NULL

File: clients/cli/connections.c [hunk 2]
```diff
@@ -4772,7 +4755,6 @@ do_connection_add (NmCli *nmc, int argc, char **argv)
 				g_string_printf (nmc->return_text, _("Error: 'save': %s."),
```
[the remark this answers is not in the document]
Comment (thom311) [reply clients/cli/connections.c]: Right. Will fix

**Comment by bengal** [comment]
See a couple of crashes caused by this branch:

https://bugzilla.redhat.com/show_bug.cgi?id=1573839
https://bugzilla.redhat.com/show_bug.cgi?id=1573840

Please close them if already solved.

**Comment by thom311** [comment]
I dropped parts of the branch for now. Remaining are the bits that are ready to merge.

**Comment by bengal** [comment]
LGTM

**Comment by thom311** [comment]
part 1 got merged as https://github.com/NetworkManager/NetworkManager/commit/2ef24490a3712ffca0faf86aecf56abcd43a1276

**Pull request merged**